lib/order.rb

```diff
@@ -1,18 +1,80 @@
+require 'csv'
+require 'awesome_print'
+
 module Grocery
   class Order
     attr_reader :id, :products
-
-    def initialize(id, products)
+    @@orders = []
+    def initialize(id, collection)
       @id = id
-      @products = products
+      @products = collection
+    end
+
+    def self.all
+      if @@orders.any?
+        return @@orders
+      end
+
+      #order = {id => {item:price, item:price}}
+      CSV.open("./support/orders.csv", "r").each do |line|
+        shopping_list = {}
+        id = line[0].to_i
+        list = line[1]
+        list = list.split(";") #now this is an array
+        list.each do |element_pair|
+          item = element_pair.split(":")
+          shopping_list[item[0]] = item[1]
+        end
```

Comment:
A lot of people didn't write code like this, but I'm so excited/happy that you did!